core/src/main/java/com/google/cloud/sql/core/InstanceCheckingTrustManger.java

```diff
@@ -92,6 +95,62 @@ private void checkCertificateChain(X509Certificate[] chain) throws CertificateEx
       throw new CertificateException("Subject is missing");
     }
 
+    if (instanceMetadata.isCasManagedCertificate()) {
+      checkSan(chain);
+    } else {
+      checkCn(chain);
+    }
+  }
+
+  private void checkSan(X509Certificate[] chain) throws CertificateException {
+    List<String> sans = getSans(chain[0]);
+    String dns = instanceMetadata.getDnsName();
+    if (dns == null || dns.isEmpty()) {
+      throw new CertificateException(
+          "Instance metadata for " + instanceMetadata.getInstanceName() + " has an empty dnsName");
+    }
+    for (String san : sans) {
+      if (san.equalsIgnoreCase(dns)) {
+        return;
+      }
+    }
+    throw new CertificateException(
+        "Server certificate does not contain expected name '"
+            + instanceMetadata.getInstanceName()
+            + "' for "
+            + instanceMetadata.getInstanceName());
+  }
+
+  private List<String> getSans(X509Certificate cert) throws CertificateException {
+    ArrayList<String> names = new ArrayList<>();
+
+    Collection<List<?>> sanAsn1Field = cert.getSubjectAlternativeNames();
+    if (sanAsn1Field == null) {
+      return names;
+    }
+
+    for (List item : sanAsn1Field) {
+      Integer type = (Integer) item.get(0);
+      if (type == 0 || type == 2) {
```

Comment:
Can we add a comment here for the 0 and 2? I assume that `0` is the `otherName` and `2` is the `dNSName` according to the RFC 5280.

Reply:
You are correct. I added a comment to clarify.